src/code/home/dialog.asm

```diff
@@ -672,22 +672,40 @@ ENDC
 .dialogBoxFull
     jp   IncrementDialogStateAndReturn            ; $268E: $C3 $85 $24
 
-data_2691::
-    db $22, $42                                   ; $2691
-
-data_2693::
-    db $98, $99                                   ; $2693
-
-; Handle a break in the dialog. Checks the NEXT character
-; to see if it's more text (if so, display arrow and
-; allow scrolling), or if it's a terminating character
-; ("@" or "<ask>"), which allows terminators to lie beyond
-; the maximum line length (otherwise a line of 16 characters
-; followed by "@" would print an empty line).
+DialogBoxOrigin::
+    ; Background tile map address of the beginning of the
+    ; text in a dialog box (one line above regular text, to
+    ; make room for diacritics?)
+.low
+    db   $22 ; top
+    db   $42 ; bottom
+.high
+    db   $98 ; top
+    db   $99 ; bottom
+
+; Handle a break in the dialog, ie. the dialog box is full but
+; there is more text left to print (wDialogState == DIALOG_BREAK)
+; 
+; This routine checks the next character to see if it's more
+; text, and if so, display arrow and wait for the player to press A.
+;
+; If the next character instead is a terminating character like
+; "@" or "<ask>", then we don't have an actual break in the dialog.
+; This allows terminators to lie beyond the maximum line
+; length of 16 characters without starting a new line; otherwise
+; a final line of exactly 16 characters followed by "@" would print
+; an extra empty line.
+;
+; However, there's a potential bug here; the check for the terminator
+; is done after checking if two full lines of characters have been
+; printed, so it only triggers on even-numbered lines. Dialog ending
+; on an odd-numbered line will still print an empty line at the end.
 DialogBreakHandler::
+    ; Bug: This check should be done after the next two
```

Comment:
I think @Xkeeper0 started tagging potential bugs with `@bug` (as several decomp or disassembly projects do).

```suggestion
    ; @bug: This check should be done after the next two
```

Reply:
Smart - I know a couple of other bugs that we should probably tag like that then.